src/env.ts

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+/* eslint-disable no-process-exit */
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+import {
+  installDependencies,
+  initPackageJsonIfNotExists,
+  updatePackageJsonDependencies,
+  extractPathsWithVersion,
+  updatePackageJsonProperties,
+} from './util/npm';
+import {parseIfEnvArgs} from './util/args';
+import {logger} from './util/logger';
+
+import {load} from './lib/load';
+
+import {CONFIG} from './config';
+
+import {EnvironmentOptions} from './types/if-env';
+
+const {IF_ENV} = CONFIG;
+const {
+  SUCCESS_MESSAGE,
+  FAILURE_MESSAGE,
+  FAILURE_MESSAGE_TEMPLATE,
+  FAILURE_MESSAGE_DEPENDENCIES,
+} = IF_ENV;
+
+const IfEnv = async () => {
+  const commandArgs = await parseIfEnvArgs();
+  const options: EnvironmentOptions = {
+    folderPath: process.env.CURRENT_DIR || process.cwd(),
+    install: !!commandArgs.install,
+    dependencies: {},
+    cwd: !!commandArgs.cwd,
+  };
+
+  if (commandArgs && commandArgs.manifest) {
+    const {folderPath, install, dependencies} =
+      await getOptionsFromArgs(commandArgs);
+    options.folderPath = commandArgs.cwd ? options.folderPath : folderPath;
+    options.install = !!install;
+    options.dependencies = {...dependencies};
+  }
+
+  await initializeAndInstallLibs(options);
+
+  if (!commandArgs || !commandArgs.manifest) {
+    await addTemplateManifest(options.folderPath);
+  }
+
+  console.log(SUCCESS_MESSAGE);
+  process.exit(0);
+};
+
+/**
+ * Gets the folder path of manifest file, dependencies from manifest file and install argument from the given arguments.
+ */
+const getOptionsFromArgs = async (commandArgs: {
+  manifest: string;
+  install: boolean | undefined;
+}) => {
+  const {manifest: manifestPath, install} = commandArgs;
```

Comment:
to escape renaming, command args function can return manifestPath instead of manifest